compiler/rustc_passes/src/dead.rs

```diff
@@ -677,30 +678,94 @@ impl<'tcx> DeadVisitor<'tcx> {
         let inherent_impls = self.tcx.inherent_impls(def_id);
         for &impl_did in inherent_impls.iter() {
             for item_did in self.tcx.associated_item_def_ids(impl_did) {
-                if let Some(def_id) = item_did.as_local() {
-                    if self.live_symbols.contains(&def_id) {
-                        return true;
-                    }
+                if let Some(def_id) = item_did.as_local()
+                    && self.live_symbols.contains(&def_id)
+                {
+                    return true;
                 }
             }
         }
         false
     }
 
-    fn warn_dead_code(
+    fn store_dead_field_or_variant(
         &mut self,
         id: hir::HirId,
         span: rustc_span::Span,
         name: Symbol,
+    ) {
+        let field_or_variant_local_def_id = |parent_node, node| -> Option<(LocalDefId, bool)> {
+            match (parent_node, node) {
+                // struct/union fields
+                (
+                    Node::Item(hir::Item {
+                        kind:
+                            hir::ItemKind::Struct(hir::VariantData::Struct(..), ..)
```

Comment:
Why not `Tuple` fields too?

Reply:
That is because rustc does not warn about tuple fields (positional fields) like the following if I am not wrong.

https://github.com/rust-lang/rust/blob/20ad8209288896259cd1829d2b057bc017315fce/compiler/rustc_passes/src/dead.rs#L653